R/utilities-simulation-results.R

```diff
@@ -83,6 +83,41 @@ getOutputValues <- function(simulationResults, quantitiesOrPaths, individualIds
   return(output)
 }
 
+#' @export
+getOutputValuesTLF <- function(simulationResults, quantitiesOrPaths, individualIds = NULL) {
+  validateIsOfType(simulationResults, SimulationResults)
+  quantitiesOrPaths <- c(quantitiesOrPaths)
+  validateIsOfType(quantitiesOrPaths, c("Quantity", "character"))
+  validateIsNumeric(individualIds, nullAllowed = TRUE)
+
+  # If quantities are passed, get their paths.
+  paths <- quantitiesOrPaths
+  if (isOfType(quantitiesOrPaths, Quantity)) {
+    paths <- unlist(lapply(quantitiesOrPaths, function(x) x$consolidatePath))
+  }
+  paths <- unique(paths)
+
+  # If no specific individual ids are passed, iterate through all individuals
+  individualIds <- ifNotNull(individualIds, unique(individualIds), simulationResults$allIndividualIds)
+
+  # All time values are equal
+  timeValues <- simulationResults$timeValues
+
+  output <- NULL;
+
+  for (individualId in individualIds) {

```

Comment:
This is quite simple here. Not sure if this is not good in terms of memory consumption. I am createing one data frame per {path, individualId" that I merge into the output one